src/test/java/de/tum/in/www1/artemis/hestia/ProgrammingExerciseTaskServiceTest.java

```diff
@@ -0,0 +1,204 @@
+package de.tum.in.www1.artemis.hestia;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import de.tum.in.www1.artemis.AbstractSpringIntegrationBambooBitbucketJiraTest;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.ProgrammingExerciseTestCase;
+import de.tum.in.www1.artemis.domain.hestia.CodeHint;
+import de.tum.in.www1.artemis.domain.hestia.ProgrammingExerciseTask;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.repository.hestia.CodeHintRepository;
+import de.tum.in.www1.artemis.repository.hestia.ProgrammingExerciseTaskRepository;
+import de.tum.in.www1.artemis.service.hestia.ProgrammingExerciseTaskService;
+
+public class ProgrammingExerciseTaskServiceTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    @Autowired
+    private ProgrammingExerciseTaskService programmingExerciseTaskService;
+
+    @Autowired
+    private ProgrammingExerciseTaskRepository programmingExerciseTaskRepository;
+
+    @Autowired
+    private ProgrammingExerciseRepository programmingExerciseRepository;
+
+    @Autowired
+    private CodeHintRepository codeHintRepository;
+
+    private ProgrammingExercise programmingExercise;
+
+    @BeforeEach
+    void init() {
+        database.addCourseWithOneProgrammingExerciseAndSpecificTestCases();
+        database.addUsers(2, 2, 1, 2);
+
+        programmingExercise = programmingExerciseRepository.findAll().get(0);
+        updateProblemStatement("""
+                [task][Task 1](testClass[BubbleSort])
+                [task][Task 2](testMethods[Context])
+                """);
+    }
+
+    private void updateProblemStatement(String problemStatement) {
+        programmingExercise.setProblemStatement(problemStatement);
+        programmingExercise = programmingExerciseRepository.save(programmingExercise);
+        programmingExerciseTaskService.updateTasksFromProblemStatement(programmingExercise);
+    }
+
+    @AfterEach
+    public void resetDatabase() {
+        database.resetDatabase();
+    }
+
+    @Test
+    public void testNewExercise() {
+        assertThat(programmingExerciseTaskRepository.findAll()).hasSize(2);
+        var tasks = programmingExerciseTaskRepository.findByExerciseIdWithTestCases(programmingExercise.getId());
+        assertThat(tasks).hasSize(2);
+
+        assertThat(tasks).anyMatch(programmingExerciseTask -> {
+            var testCases = programmingExerciseTask.getTestCases();
+            return "Task 1".equals(programmingExerciseTask.getTaskName()) && !testCases.isEmpty()
+                    && "testClass[BubbleSort]".equals(testCases.stream().findFirst().get().getTestName());
```

Comment:
This is repeated a few times. Can you extract it to a helper method that takes the task, the expected name, and the expected testName as parameters?